File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;
import java.io.*;
class Solution {
    public boolean solution(String[] phone_book) {
        Arrays.sort(phone_book);
        
        for(int i = 0; i < phone_book.length-1; i++) {
            String prefix = phone_book[i];
            String next = phone_book[i+1];
            
            if(next.startsWith(prefix)) return false;

            
            
        }
        
        return true;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    phone_book.sort()
    
    i = 0
    while(i < len(phone_book)-1):
        prefix = phone_book[i]
        next = phone_book[i+1]
        if prefix == next[:len(prefix)]:
            answer = False
            break
        i += 1
    return answer

